test/test_sync.cpp

```diff
@@ -7965,6 +7965,7 @@ TEST(Sync_Dictionary)
         auto& g = tr.get_group();
         auto foos = g.add_table_with_primary_key("class_Foo", type_Int, "id");
         auto col_dict = foos->add_column_dictionary(type_Mixed, "dict");
+        auto col_dict_str = foos->add_column_dictionary(type_String, "str_dict", true);
```

Comment:
Do we test that the nullability also makes it into the spec after sync? Not sure if `compare_groups()` tests this flag.

Reply:
It is implicitly tested. The nullability is just a bit in ColKey, which is stored in the spec. I will check `compare_groups`.

Reply:
`compare_groups` checks the nullable flag when comparing schemas - no matter what kind if column.